devolo_home_control_api/properties/settings_property.py

```diff
@@ -10,22 +10,24 @@ class SettingsProperty(Property):
     the gateway. This is to be as flexible to gateway firmware changes as possible. So if new attributes appear or old ones
     are removed, they should be handled at least in reading them. Nevertheless, a few unwanted attributes are filtered.
 
-    :param connection: Collection of instances needed to communicate with the central unit
-    :param element_uid: Element UID, something like devolo.BinarySwitch:hdm:ZWave:CBC56091/24#2
+    :param element_uid: Element UID, something like acs.hdm:ZWave:CBC56091/24
+    :param setter: Method to call on setting the state
     :param **kwargs: Any setting, that shall be available in this object
     """
 
-    def __init__(self, connection: Dict, element_uid: str, **kwargs: Any):
+    def __init__(self, element_uid: str, setter: Callable, **kwargs: Any):
         if not element_uid.startswith(("acs", "bas", "bss", "cps", "gds", "lis", "mas",
                                        "mss", "ps", "sts", "stmss", "trs", "vfs")):
             raise WrongElementError()
 
-        super().__init__(connection=connection, element_uid=element_uid)
-        for key, value in kwargs.items():
-            setattr(self, key, value)
+        super().__init__(element_uid=element_uid)
+        self._setter = setter
 
         if element_uid.startswith("gds"):
-            self.zone = self._gateway.zones[self.zone_id]
+            self.zone = kwargs.pop("zones")[kwargs["zone_id"]]
```

Comment:
Is a KeyError possible here?

Reply:
Yes it is. I took care of that.

Reply:
Can be simplified like this:

`if element_uid.startswith("gds") and all(e in kwargs for e in ["zones", "zone_id"]):`

Reply:
"Done"